src/workerd/api/worker-loader-test.js

```diff
@@ -0,0 +1,135 @@
+import { DurableObject } from 'cloudflare:workers';
+import assert from 'node:assert';
+
+export let basics = {
+  async test(ctrl, env, ctx) {
+    let worker = env.loader.get('basics', () => {
```

Comment:
these tests can be expanded.

* The loader function return value is actually converted into a promise internally but these are all sync functions. Having a test variation that explicitly returns a promise that awaits some i/o (simple timer) would be good so that we're not always testing with synchronously resolved promises.
* None of the current test cases appear to test what happens if the loader callback throws or returns invalid data.
* Having a case where a worker tries loading a worker twice just to verify it works as expected ... e.g. `let w1 = env.loader.get('basics', ...); let w2 = env.loader.get('basics', ...); await Promise.all([w2.getEntrypoint().greet('...'), w1.getEntrypoint().greet('...')]);` (note that flipping the order of the getEntrypoint calls is intentional there).